src/platform/ESP32/nimble/BLEManagerImpl.cpp

```diff
@@ -0,0 +1,1074 @@
+    }
+
+    // Configure the BLE device name.
+    err = ble_svc_gap_device_name_set(mDeviceName);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "ble_svc_gap_device_name_set() failed: %s", ErrorStr(err));
+        ExitNow();
+    }
+
+    advDataVersionDiscriminator = (discriminator | (CHIP_ADV_VERSION << 12));
+
+    memset(advData, 0, sizeof(advData));
+    advData[index++] = 0x02;                             // length
+    advData[index++] = CHIP_ADV_DATA_TYPE_FLAGS;         // AD type : flags
+    advData[index++] = CHIP_ADV_DATA_FLAGS;              // AD value
+    advData[index++] = 0x06;                             // length
+    advData[index++] = CHIP_ADV_DATA_TYPE_SERVICE_DATA;  // AD type: (Service Data - 16-bit UUID)
+    advData[index++] = ShortUUID_CHIPoBLEService[0];     // AD value
+    advData[index++] = ShortUUID_CHIPoBLEService[1];     // AD value
+    advData[index++] = CHIP_BLE_OPCODE;                  // Used to differentiate from operational CHIP Ble adv
+    advData[index++] = advDataVersionDiscriminator >> 8; // Bits[15:12] == version - Bits[11:0] Discriminator
+    advData[index++] = advDataVersionDiscriminator & 0x00FF;
+
+    // Construct the Chip BLE Service Data to be sent in the scan response packet.
+    err = ble_gap_adv_set_data(advData, sizeof(advData));
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "ble_gap_adv_set_data failed: %s %d", ErrorStr(err), advDataVersionDiscriminator);
+        ExitNow();
+    }
+
+exit:
+    return err;
+}
+
+void BLEManagerImpl::HandleRXCharWrite(struct ble_gatt_char_context * param)
+{
+    CHIP_ERROR err     = CHIP_NO_ERROR;
+    PacketBuffer * buf = NULL;
+    uint16_t data_len  = 0;
+
+    ESP_LOGI(TAG, "Write request received for CHIPoBLE RX characteristic con %u %u", param->conn_handle, param->attr_handle);
+
+    // Copy the data to a PacketBuffer.
+    buf = PacketBuffer::New(0);
+    VerifyOrExit(buf != NULL, err = CHIP_ERROR_NO_MEMORY);
+    data_len = OS_MBUF_PKTLEN(param->ctxt->om);
+    VerifyOrExit(buf->AvailableDataLength() >= data_len, err = CHIP_ERROR_BUFFER_TOO_SMALL);
+    ble_hs_mbuf_to_flat(param->ctxt->om, buf->Start(), data_len, NULL);
+    buf->SetDataLength(data_len);
+
+    // Post an event to the Chip queue to deliver the data into the Chip stack.
+    {
+        ChipDeviceEvent event;
+        event.Type                        = DeviceEventType::kCHIPoBLEWriteReceived;
+        event.CHIPoBLEWriteReceived.ConId = param->conn_handle;
+        event.CHIPoBLEWriteReceived.Data  = buf;
+        PlatformMgr().PostEvent(&event);
+        buf = NULL;
+    }
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "HandleRXCharWrite() failed: %s", ErrorStr(err));
+    }
+    if (buf != NULL)
+    {
+        PacketBuffer::Free(buf);
+    }
+}
+
+void BLEManagerImpl::HandleTXCharRead(struct ble_gatt_char_context * param)
+{
+    /* XXX Not supported */
```

Comment:
Does the XXX means it will be supported at some point ? Otherwise if that's something we don't plan to support we can get rid of the XXX.

Reply:
Done!!